.github/workflows/reusable-test.yml

```diff
@@ -0,0 +1,51 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+
+name: CI
+
+on:
+  workflow_call:
+    inputs:
+      os-variant:
+        default: 'ubuntu-20.04'
+        type: string
+      build-config:
+        default: 'dev-env'
+        type: string
+      python-version:
+        default: '3.9'
+        type: string
+      coverage-report:
+        default: false
+        type: boolean
+
+jobs:
+  test:
+    name: tests ${{ inputs.os-variant }}, py${{ inputs.python-version }}
+    runs-on: ${{ inputs.os-variant }}
+    defaults:
+      run:
+        shell: bash -l {0}
+    strategy:
+      fail-fast: false
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+      - name: Setup conda environment
+        uses: mamba-org/provision-with-micromamba@main
+        with:
+          micromamba-version: 1.0.0
+          environment-file: docs/environments/conda/${{ inputs.build-config }}.yaml
```

Comment:
Are you planning on having multiple different environments? If so, for what purpose?

Reply:
I don't know, I copied and pasted this from other repositories and 
I thought it'd be better to keep it re-usuable for other repositories in case we want to merge everything...?